File: zscheduler-server/src/main/java/com/github/slablock/zscheduler/server/domain/ScheduleExpression.java
package com.github.slablock.zscheduler.server.domain;

public class ScheduleExpression {

    private Integer scheduleType;
    private String expression;

    public Integer getScheduleType() {
        return scheduleType;
    }

    public void setScheduleType(Integer scheduleType) {
        this.scheduleType = scheduleType;
    }

    public String getExpression() {
        return expression;
    }

    public void setExpression(String expression) {
        this.expression = expression;
    }
}

File: zscheduler-core/src/main/scala/com/github/slablock/zcheduler/core/domain/ScheduleExpression.scala
package com.github.slablock.zcheduler.core.domain

import org.joda.time.DateTime

abstract class ScheduleExpression(expression: String) extends Expression {

  def getExpression: String = expression

  def getTimeBefore(dateTime: DateTime): Option[DateTime]

  def getTimeAfter(dateTime: DateTime): Option[DateTime]

}
